modules/animation/src/animationcontroller.cpp

```diff
@@ -135,105 +112,79 @@ AnimationController::AnimationController(Animation& animation, InviwoApplication
                        0)
     , renderWindow("RenderFirstLastTime", "Window", 0, 10, 0, 1e5, 1, 0.0,
                    InvalidationLevel::InvalidOutput, PropertySemantics::Text)
-    , renderSizeMode("RenderSizeOptions", "Size",
-                     {{"CurrentCanvas", "Use current settings of canvases", 0},
-                      {"720p", "720p for all canvases", 1},
-                      {"1080p", "1080p for all canvases", 2},
-                      {"CustomSize", "User-defined resolution for all canvases", 3}},
-                     0)
-    , renderSize("RenderSize", "Pixels", ivec2(1024), ivec2(1), ivec2(20000), ivec2(256),
-                 InvalidationLevel::InvalidOutput, PropertySemantics::Text)
-    , renderAspectRatio("RenderSizeAspectRatio", "Aspect Ratio",
-                        {{"Ignore", "Ignore aspect ratio", 0},
-                         {"KeepInside", "Keep aspect ratio within given resolution", 1},
-                         {"KeepEnlarge", "Keep aspect ratio exceeding given resolution", 2}},
-                        1)
     , renderLocation("RenderLocationDir", "Directory")
     , renderBaseName("RenderLocationBaseName", "Base Name")
-    , renderImageExtension("RenderImageExtension", "Type", imageExts(app),
-                           imageExtIndex(app, defaultImageExt))
-    , renderNumFrames("RenderNumFrames", "# Frames", 100, 2, 1000000, 1,
-                      InvalidationLevel::InvalidOutput, PropertySemantics::Text)
-    , renderAction("RenderAction", "Render")
-    , renderActionStop("RenderActionStop", "Stop")
-    , controlOptions("ControlOptions", "Control Track")
-    , controlInsertPauseFrame("ControlInsertPauseFrame", "Add Control Track")
+    , writer("writer", "Type",
+             [&]() {
+                 OptionPropertyState<FileExtension> state;
+                 const auto exts = util::getDataWriterFactory(app)->getExtensionsForType<Layer>();
+                 std::transform(
+                     exts.begin(), exts.end(), std::back_inserter(state.options),
+                     [](const auto& ext) -> OptionPropertyOption<FileExtension> { return ext; });
+                 auto it = std::find_if(exts.begin(), exts.end(),
+                                        [&](auto& e) { return e.extension_ == "png"; });
+                 if (it != exts.end()) {
+                     state.selectedIndex = std::distance(exts.begin(), it);
+                 }
+                 return state;
+             }())
+    , renderFPS("renderFPS", "Frames per Second", 100, 2, 1000000, 1,
```

Comment:
Ranges need to be adapted because the meaning of this property changed
```suggestion
    , renderFPS("renderFPS", "Frames per Second", 24.0,
                      {1, ConstraintBehavior::Immutable},
                      {60, ConstraintBehavior::Editable}, 1.0
```